db/snapshot_checker.h

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+#include "rocksdb/types.h"
+
+namespace rocksdb {
+
+class WritePreparedTxnDB;
+
+// Callback class created by WritePreparedTxnDB to check if a key
+// is visible by a snapshot.
+class SnapshotChecker {
+ public:
+  explicit SnapshotChecker(WritePreparedTxnDB* txn_db);
+
+  bool IsInSnapshot(SequenceNumber sequence,
+                    SequenceNumber snapshot_sequence) const;
+
+ private:
+#ifndef ROCKSDB_LITE
+  const WritePreparedTxnDB* const txn_db_;
+#endif  // !ROCKSDB_LITE
+};
+
+#ifdef ROCKSDB_LITE
+SnapshotChecker::SnapshotChecker(WritePreparedTxnDB* txn_db) {}
+
+bool SnapshotChecker::IsInSnapshot(SequenceNumber sequence,
+                                   SequenceNumber snapshot_sequence) const {
+  return sequence <= snapshot_sequence;
```

Comment:
This is confusing. If we do not support WritePrepared in ROCKSDB_LITE, then SnapshotChecker should be set to nullptr. This function should never be called in that case so it should throw an exception if it is called.

Reply:
snapshot_checker is not being used in LITE mode. The implementation here is just a place holder.

Reply:
I understand that but providing an implementation that is never used could be confusing. Return true all the time with a proper in-line comment would make it more clear.